packages/expo-cli/src/commands/utils/__tests__/validateDependenciesVersions-test.ts

```diff
@@ -73,4 +75,44 @@ describe(validateDependenciesVersionsAsync, () => {
       false
     );
   });
+
+  it('resolves to true when installed package uses "exports"', async () => {
+    const packageJsonPath = path.join(projectRoot, 'node_modules/firebase/package.json');
+
+    vol.fromJSON({
+      [packageJsonPath]: JSON.stringify({
+        version: '9.1.0',
+        exports: {
+          './analytics': {
+            node: {
+              require: './analytics/dist/index.cjs.js',
+              import: './analytics/dist/index.mjs',
+            },
+            default: './analytics/dist/index.esm.js',
+          },
+        },
+      }),
+    });
+
+    // Manually trigger the Node import error for "exports".
+    // This isn't triggered by memfs, or our mock, that's why we need to do it manually.
+    // see: https://github.com/expo/expo-cli/pull/3878
+    (resolveFrom as jest.MockedFunction<typeof resolveFrom>).mockImplementationOnce(() => {
```

Comment:
So this was a pain to implement, we have to trigger the warning manually because we mock the whole system. We use memfs and some mocked/fake version of resolve from. At least this fakes the real world now 😄  (it actually tests this fix now)